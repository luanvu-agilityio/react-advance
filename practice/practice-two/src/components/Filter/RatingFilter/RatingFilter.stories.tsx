import type { Meta, StoryObj } from '@storybook/react'
import { useState } from 'react'
import { RatingFilter } from './RatingFilter'
import type { RatingProps } from 'types/Filter'

const meta: Meta<typeof RatingFilter> = {
  title: 'Components/Filter/RatingFilter',
  component: RatingFilter,
  parameters: {
    layout: 'centered',
    backgrounds: {
      default: 'light',
    },
  },
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof RatingFilter>

/**
 * Default RatingFilter with various rating options
 */
export const Default: Story = {
  args: {
    ratings: [
      { rating: 5, selected: false },
      { rating: 4, selected: true },
      { rating: 3, selected: false },
      { rating: 2, selected: false },
      { rating: 1, selected: false },
    ],
    onRatingChange: (index) => {
      console.log(`Rating at index ${index} clicked`)
    },
  },
}

/**
 * RatingFilter with no selected ratings
 */
export const NoSelectedRatings: Story = {
  args: {
    ratings: [
      { rating: 5, selected: false },
      { rating: 4, selected: false },
      { rating: 3, selected: false },
      { rating: 2, selected: false },
      { rating: 1, selected: false },
    ],
    onRatingChange: (index) => {
      console.log(`Rating at index ${index} clicked`)
    },
  },
}

/**
 * RatingFilter with multiple selected ratings
 */
export const MultipleSelectedRatings: Story = {
  args: {
    ratings: [
      { rating: 5, selected: true },
      { rating: 4, selected: true },
      { rating: 3, selected: false },
      { rating: 2, selected: true },
      { rating: 1, selected: false },
    ],
    onRatingChange: (index) => {
      console.log(`Rating at index ${index} clicked`)
    },
  },
}

/**
 * Interactive RatingFilter that demonstrates state changes
 */
export const Interactive: Story = {
  render: function InteractiveRatingFilter() {
    const [ratings, setRatings] = useState<RatingProps[]>([
      { rating: 5, selected: false },
      { rating: 4, selected: false },
      { rating: 3, selected: false },
      { rating: 2, selected: false },
      { rating: 1, selected: false },
    ])

    const handleRatingChange = (index: number) => {
      const newRatings = [...ratings]
      newRatings[index].selected = !newRatings[index].selected
      setRatings(newRatings)
    }

    return (
      <RatingFilter ratings={ratings} onRatingChange={handleRatingChange} />
    )
  },
}

/**
 * RatingFilter with a single rating option
 */
export const SingleRating: Story = {
  args: {
    ratings: [{ rating: 5, selected: true }],
    onRatingChange: () => {},
  },
}

/**
 * RatingFilter with empty ratings array (should not render stars)
 */
export const EmptyRatings: Story = {
  args: {
    ratings: [],
    onRatingChange: () => {},
  },
}
