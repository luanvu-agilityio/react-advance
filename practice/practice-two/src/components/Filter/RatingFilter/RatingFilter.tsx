import { CheckIcon } from 'lucide-react'
import {
  SectionTitle,
  RatingSection,
  StarItems,
  CheckboxContainer,
  CheckboxRoot,
  CheckboxIndicator,
  StarsWrapper,
  StarIcon,
<<<<<<< HEAD
=======
  RatingLabel,
>>>>>>> 7a35d9791a5da6fe80ff0a8541efaf78233de04d
} from '../FilterStyles'
import type { RatingProps } from 'types/Filter'

interface RatingFilterProps {
  ratings: RatingProps[]
  onRatingChange: (index: number) => void
}

export const RatingFilter = ({
  ratings,
  onRatingChange,
}: RatingFilterProps) => {
  return (
    <RatingSection>
      <SectionTitle>Rating</SectionTitle>
      <StarItems>
        {ratings.map((rating, index) => (
          <CheckboxContainer key={index}>
            <CheckboxRoot
              checked={rating.selected}
              onCheckedChange={() => onRatingChange(index)}
              id={`rating-${index}`}
            >
              <CheckboxIndicator>
                <CheckIcon strokeWidth={2} style={{ paddingTop: '5px' }} />
              </CheckboxIndicator>
            </CheckboxRoot>
            <StarsWrapper>
              {[...Array(5)].map((_, i) => (
                <StarIcon key={i} $filled={i < rating.rating} />
              ))}
            </StarsWrapper>
          </CheckboxContainer>
        ))}
      </StarItems>
    </RatingSection>
  )
}
