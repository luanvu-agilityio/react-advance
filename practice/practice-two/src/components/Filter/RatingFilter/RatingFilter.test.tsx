import { render, screen, fireEvent } from '@testing-library/react'
import { RatingFilter } from './RatingFilter'
import type { RatingProps, StarRating } from 'types/Filter'

// Only mock the CheckIcon component, not the Star component since we're not using it anymore
jest.mock('lucide-react', () => ({
  CheckIcon: () => <div data-testid="check-icon">✓</div>,
}))

// Mock the styled-components to control their rendering in tests
jest.mock('../FilterStyles', () => {
  const actual = jest.requireActual('../FilterStyles')
  return {
    ...actual,
    // Mock the StarIcon component to return a testable element
    StarIcon: ({ $filled }: { $filled: boolean }) => (
      <div data-testid="star-icon" data-filled={$filled ? 'true' : 'false'}>
        {$filled ? '★' : '☆'}
      </div>
    ),
  }
})

describe('RatingFilter', () => {
  const mockRatings: RatingProps[] = [
    { rating: 5, selected: false },
    { rating: 4, selected: true },
    { rating: 3, selected: false },
    { rating: 2, selected: false },
    { rating: 1, selected: false },
  ]

  const mockOnRatingChange = jest.fn()

  beforeEach(() => {
    mockOnRatingChange.mockClear()
  })

  // SNAPSHOT TESTS FOR RENDERING
  describe('Rendering', () => {
    it('renders different component states correctly', () => {
      // Default state with mixed selections
      const { container: defaultContainer } = render(
        <RatingFilter
          ratings={mockRatings}
          onRatingChange={mockOnRatingChange}
        />
      )
      expect(defaultContainer).toMatchSnapshot('default-state')

      // All ratings selected
      const allSelected = mockRatings.map((rating) => ({
        ...rating,
        selected: true,
      }))
      const { container: allSelectedContainer } = render(
        <RatingFilter
          ratings={allSelected}
          onRatingChange={mockOnRatingChange}
        />
      )
      expect(allSelectedContainer).toMatchSnapshot('all-selected')

      // No ratings selected
      const noneSelected = mockRatings.map((rating) => ({
        ...rating,
        selected: false,
      }))
      const { container: noneSelectedContainer } = render(
        <RatingFilter
          ratings={noneSelected}
          onRatingChange={mockOnRatingChange}
        />
      )
      expect(noneSelectedContainer).toMatchSnapshot('none-selected')

      // Empty ratings array
      const { container: emptyContainer } = render(
        <RatingFilter ratings={[]} onRatingChange={mockOnRatingChange} />
      )
      expect(emptyContainer).toMatchSnapshot('empty-ratings')

      // Only one rating option
      const singleRating = [{ rating: 3 as StarRating, selected: true }]
      const { container: singleRatingContainer } = render(
        <RatingFilter
          ratings={singleRating}
          onRatingChange={mockOnRatingChange}
        />
      )
      expect(singleRatingContainer).toMatchSnapshot('single-rating')
    })
  })

  // Add a new test for the stars rendering
  it('renders the correct number of filled and empty stars', () => {
    render(
      <RatingFilter ratings={mockRatings} onRatingChange={mockOnRatingChange} />
    )

    // Get all star icons
    const starIcons = screen.getAllByTestId('star-icon')

    // For the 5-star rating (first rating in our mock)
    // All 5 stars should be filled
    expect(starIcons[0]).toHaveAttribute('data-filled', 'true')
    expect(starIcons[1]).toHaveAttribute('data-filled', 'true')
    expect(starIcons[2]).toHaveAttribute('data-filled', 'true')
    expect(starIcons[3]).toHaveAttribute('data-filled', 'true')
    expect(starIcons[4]).toHaveAttribute('data-filled', 'true')

    // For the 4-star rating (second rating in our mock)
    // First 4 stars should be filled, 5th star empty
    expect(starIcons[5]).toHaveAttribute('data-filled', 'true')
    expect(starIcons[6]).toHaveAttribute('data-filled', 'true')
    expect(starIcons[7]).toHaveAttribute('data-filled', 'true')
    expect(starIcons[8]).toHaveAttribute('data-filled', 'true')
    expect(starIcons[9]).toHaveAttribute('data-filled', 'false')
  })

  // INTERACTIVE BEHAVIOR TESTS
  describe('Interactive behavior', () => {
    it('calls onRatingChange when clicking on a checkbox', () => {
      render(
        <RatingFilter
          ratings={mockRatings}
          onRatingChange={mockOnRatingChange}
        />
      )

      // Find the checkbox elements
      const checkboxes = screen.getAllByRole('checkbox')

      // Click on the first checkbox (5 Stars & Up)
      fireEvent.click(checkboxes[0])
      expect(mockOnRatingChange).toHaveBeenCalledWith(0)

      mockOnRatingChange.mockClear()

      // Click on the last checkbox (1 Star & Up)
      fireEvent.click(checkboxes[4])
      expect(mockOnRatingChange).toHaveBeenCalledWith(4)
    })

    it('allows toggling an already selected rating', () => {
      render(
        <RatingFilter
          ratings={mockRatings}
          onRatingChange={mockOnRatingChange}
        />
      )

      // 4 Stars & Up is already selected in our mock data (index 1)
      const selectedRatingCheckbox = screen.getAllByRole('checkbox')[1]

      // Verify it's initially checked before clicking
      expect(selectedRatingCheckbox).toBeChecked()

      // Click to toggle it off
      fireEvent.click(selectedRatingCheckbox)
      expect(mockOnRatingChange).toHaveBeenCalledWith(1)
    })
  })
})
