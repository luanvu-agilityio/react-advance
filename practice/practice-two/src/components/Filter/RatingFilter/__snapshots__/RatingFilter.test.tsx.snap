// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingFilter Rendering renders different component states correctly: all-selected 1`] = `
<div>
  <div
    class="sc-jScdur lcnHry"
  >
    <h2
      class="sc-fYsHOw gpnbOD"
    >
      Rating
    </h2>
    <div
      class="sc-dcKlJK hqQole"
    >
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="true"
          class="sc-cXawGu VtGFV"
          data-state="checked"
          id="rating-0"
          role="checkbox"
          type="button"
          value="on"
        >
          <span
            class="sc-jJAtPt dVvwCd"
            data-state="checked"
            style="pointer-events: none;"
          >
            <div
              data-testid="check-icon"
            >
              ✓
            </div>
          </span>
        </button>
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="true"
          class="sc-cXawGu VtGFV"
          data-state="checked"
          id="rating-1"
          role="checkbox"
          type="button"
          value="on"
        >
          <span
            class="sc-jJAtPt dVvwCd"
            data-state="checked"
            style="pointer-events: none;"
          >
            <div
              data-testid="check-icon"
            >
              ✓
            </div>
          </span>
        </button>
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="true"
          class="sc-cXawGu VtGFV"
          data-state="checked"
          id="rating-2"
          role="checkbox"
          type="button"
          value="on"
        >
          <span
            class="sc-jJAtPt dVvwCd"
            data-state="checked"
            style="pointer-events: none;"
          >
            <div
              data-testid="check-icon"
            >
              ✓
            </div>
          </span>
        </button>
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="true"
          class="sc-cXawGu VtGFV"
          data-state="checked"
          id="rating-3"
          role="checkbox"
          type="button"
          value="on"
        >
          <span
            class="sc-jJAtPt dVvwCd"
            data-state="checked"
            style="pointer-events: none;"
          >
            <div
              data-testid="check-icon"
            >
              ✓
            </div>
          </span>
        </button>
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="true"
          class="sc-cXawGu VtGFV"
          data-state="checked"
          id="rating-4"
          role="checkbox"
          type="button"
          value="on"
        >
          <span
            class="sc-jJAtPt dVvwCd"
            data-state="checked"
            style="pointer-events: none;"
          >
            <div
              data-testid="check-icon"
            >
              ✓
            </div>
          </span>
        </button>
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RatingFilter Rendering renders different component states correctly: default-state 1`] = `
<div>
  <div
    class="sc-jScdur lcnHry"
  >
    <h2
      class="sc-fYsHOw gpnbOD"
    >
      Rating
    </h2>
    <div
      class="sc-dcKlJK hqQole"
    >
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="false"
          class="sc-cXawGu VtGFV"
          data-state="unchecked"
          id="rating-0"
          role="checkbox"
          type="button"
          value="on"
        />
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="true"
          class="sc-cXawGu VtGFV"
          data-state="checked"
          id="rating-1"
          role="checkbox"
          type="button"
          value="on"
        >
          <span
            class="sc-jJAtPt dVvwCd"
            data-state="checked"
            style="pointer-events: none;"
          >
            <div
              data-testid="check-icon"
            >
              ✓
            </div>
          </span>
        </button>
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="false"
          class="sc-cXawGu VtGFV"
          data-state="unchecked"
          id="rating-2"
          role="checkbox"
          type="button"
          value="on"
        />
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="false"
          class="sc-cXawGu VtGFV"
          data-state="unchecked"
          id="rating-3"
          role="checkbox"
          type="button"
          value="on"
        />
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="false"
          class="sc-cXawGu VtGFV"
          data-state="unchecked"
          id="rating-4"
          role="checkbox"
          type="button"
          value="on"
        />
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RatingFilter Rendering renders different component states correctly: empty-ratings 1`] = `
<div>
  <div
    class="sc-jScdur lcnHry"
  >
    <h2
      class="sc-fYsHOw gpnbOD"
    >
      Rating
    </h2>
    <div
      class="sc-dcKlJK hqQole"
    />
  </div>
</div>
`;

exports[`RatingFilter Rendering renders different component states correctly: none-selected 1`] = `
<div>
  <div
    class="sc-jScdur lcnHry"
  >
    <h2
      class="sc-fYsHOw gpnbOD"
    >
      Rating
    </h2>
    <div
      class="sc-dcKlJK hqQole"
    >
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="false"
          class="sc-cXawGu VtGFV"
          data-state="unchecked"
          id="rating-0"
          role="checkbox"
          type="button"
          value="on"
        />
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="false"
          class="sc-cXawGu VtGFV"
          data-state="unchecked"
          id="rating-1"
          role="checkbox"
          type="button"
          value="on"
        />
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="false"
          class="sc-cXawGu VtGFV"
          data-state="unchecked"
          id="rating-2"
          role="checkbox"
          type="button"
          value="on"
        />
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="false"
          class="sc-cXawGu VtGFV"
          data-state="unchecked"
          id="rating-3"
          role="checkbox"
          type="button"
          value="on"
        />
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="false"
          class="sc-cXawGu VtGFV"
          data-state="unchecked"
          id="rating-4"
          role="checkbox"
          type="button"
          value="on"
        />
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RatingFilter Rendering renders different component states correctly: single-rating 1`] = `
<div>
  <div
    class="sc-jScdur lcnHry"
  >
    <h2
      class="sc-fYsHOw gpnbOD"
    >
      Rating
    </h2>
    <div
      class="sc-dcKlJK hqQole"
    >
      <div
        class="sc-jMsorb cyLRQD"
      >
        <button
          aria-checked="true"
          class="sc-cXawGu VtGFV"
          data-state="checked"
          id="rating-0"
          role="checkbox"
          type="button"
          value="on"
        >
          <span
            class="sc-jJAtPt dVvwCd"
            data-state="checked"
            style="pointer-events: none;"
          >
            <div
              data-testid="check-icon"
            >
              ✓
            </div>
          </span>
        </button>
        <div
          class="sc-jPkiSJ hHndaL"
        >
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="true"
            data-testid="star-icon"
          >
            ★
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
          <div
            data-filled="false"
            data-testid="star-icon"
          >
            ☆
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
